### ISSUE TITLE
function AdminController.init/1 is undefined (module AdminController is not available)

### ISSUE THREAD

**Issue by raarts** [smpallen99/elixir_survey_tutorial#1]
I think I did everything according to the tutorial, but I got: 

```
Request: GET /
** (exit) an exception was raised:
    ** (UndefinedFunctionError) function AdminController.init/1 is undefined (module AdminController is not available)
        AdminController.init(:dashboard)
        (survey) web/router.ex:1: anonymous fn/1 in Survey.Router.match_route/4
        (survey) lib/phoenix/router.ex:261: Survey.Router.dispatch/2
        (survey) web/router.ex:1: Survey.Router.do_call/2
        (survey) lib/survey/endpoint.ex:1: Survey.Endpoint.phoenix_pipeline/1
        (survey) lib/plug/debugger.ex:123: Survey.Endpoint."call (overridable 3)"/2
        (survey) lib/survey/endpoint.ex:1: Survey.Endpoint.call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:15: Plug.Adapters.Cowboy.Handler.upgrade/4
        (cowboy) src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4
```

**Comment by dmytronasyrov** [comment]
Same, do you already know how to get rid of this?

**Comment by smpallen99** [comment]
This is a really old project. It needs to be updated. I'm really backlogged on open source right now. I'll do my best to take a look at soon.

**Comment by smpallen99** [comment]
I just pushed some changes that fixes the issue. We changed the API for calling `admin_routes`. This was causing the error:

```elixir
  scope "/admin", ExAdmin do
    pipe_through :browser
    admin_routes()
  end
```

I also updated a bunch of stuff so the project works with Phoenix 1.3 and Ecto 2.1. 

You will still see a bunch of compile warnings from :speak_ex.. The macros in the menu  module required refactoring to remove using local variables outside a `if`, `case`, and `cond`.  I'll add an issue for speak_ex to track this issue.
